alchemy/src/cloudflare/api-error.ts

```diff
@@ -42,12 +42,56 @@ export class CloudflareApiError extends Error {
  * @param resourceName The name/identifier of the specific resource
  * @returns Never returns - always throws an error
  */
+/**
+ * Helper function to detect and handle OAuth token authentication limitations
+ * Provides specific guidance when OAuth tokens don't support required operations
+ *
+ * @param response The fetch Response object
+ * @param action The action being performed (e.g., "creating", "deleting")
+ * @param resourceType The type of resource being acted upon (e.g., "email routing")
+ * @returns Promise<never> if it's an OAuth error, undefined otherwise
+ */
+export async function handleOAuthError(
+  response: Response,
+  action: string,
+  resourceType: string,
+): Promise<never | undefined> {
+  const text = await response.text();
```

Comment:
This will consume the response before before the rest of the handleApiError gets to run